src/Credentials.php

```diff
@@ -63,6 +69,47 @@ private function getLWAToken()
 
      return $json['access_token'];
 
+    }
+
+    /**
+     * Request a Login with Amazon access token
+     * @see https://github.com/amzn/selling-partner-api-docs/blob/main/guides/developer-guide/SellingPartnerApiDeveloperGuide.md#step-1-request-a-login-with-amazon-access-token
+     * @return mixed
+     * @throws \Exception
+     */
+    public function getMigrationToken()
+    {
+        $knownToken = $this->loadTokenFromStorage('migration_token');
+        if (!is_null($knownToken)) {
+            return $knownToken;
+        }
+
+        $client = $this->createHttpClient([
+            'base_uri' => 'https://api.amazon.com'
+        ]);
+
+        try {
+            $requestOptions = [
+                'form_params' => [
+                    'grant_type' => 'client_credentials',
+                    'scope' => 'sellingpartnerapi::migration',
+                    'client_id' => $this->config['client_id'],
+                    'client_secret' => $this->config['client_secret']
+                ]
+            ];
+            $response = $client->post('/auth/o2/token', $requestOptions);
+        } catch (\Exception $e) {
+            //log something
+            throw $e;
+        }
+        $json = json_decode($response->getBody(), true);
+
+        $this->tokenStorege->storeToken('migration_token', [
+            'token' => $json['access_token'],
+            'expiresOn' => time() + ($this->config['access_token_longevity'] ?? 3600)
```

Comment:
NIT
Hi there, I am not familiar with this routine, but I have a question:
How we are sure that the migration token longevity is the same as the access token?

Wouldn't it be better to use a separate configuration key, or better decide it from the response in LINE 100?

Reply:
Hi @lyubo-slavilov, thanks for review :)
You are right, there is an TTL time in response, changed to using it from the response value.